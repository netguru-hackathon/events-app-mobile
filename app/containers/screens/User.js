import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components/native';
import {
  Image,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import I18n from '../../utils/translations';
import { RenderActivityIndicator } from '../shared/RenderActivityIndicator';
import colors from '../../constants/colors';

// actions
import { fetchUser } from '../../actions/users';

const Container = styled.View`
  margin: 10px;
  align-items: center;
`;

const UserAvatar = styled.Image`
  width: 120px;
  height: 120px;
  margin: 5px;
`;

const UserName = styled.Text`
  font-size: 18px;
  font-weight: bold;
`;

const UserEmail = styled.Text`
  margin-bottom: 10px;
`;

const styles = StyleSheet.create({
  icon: {
    width: 20,
    height: 20,
  },
});

class User extends Component {
  static propTypes = {
    fetchUser: PropTypes.func.isRequired,
    isStarted: PropTypes.bool.isRequired,
    isFetching: PropTypes.bool.isRequired,
    user: PropTypes.shape({
      avatarUrl: PropTypes.string,
      email: PropTypes.string,
      name: PropTypes.string,
    }),
    navigation: PropTypes.shape({
      state: PropTypes.object.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  static navigationOptions = (props) => {
    const { navigation } = props;

    return {
      tabBarLabel: I18n.t('events'),
      showIcon: true,
      tabBarIcon: ({ tintColor }) => (
        <Image
          source={require('../../img/EventsIcon.png')}
          style={[styles.icon, { tintColor }]}
        />
      ),
      headerTitle: `${navigation.state.params.item.name}`,
      headerTintColor: colors.WHITE,
      headerStyle: {
        backgroundColor: colors.BLACK_OPACITY,
      },
    };
  };

  componentWillMount() {
    this.props.fetchUser(this.props.navigation.state.params.item.id);
  }

  renderUser = () => {
    const { isStarted, isFetching, user } = this.props;

    if (isStarted && isFetching) return <RenderActivityIndicator />;
    return (
      <View>
        <UserAvatar source={{ uri: user.avatarUrl }} />
        <UserName>{user.name}</UserName>
        <UserEmail>{user.email}</UserEmail>
      </View>
    );
  }

  render() {
    return (
      <Container>
        {this.renderUser()}
      </Container>
    );
  }
}

const mapStateToProps = state => ({
  isStarted: state.user.isStarted,
  isFetching: state.user.isFetching,
  user: state.user.item,
});

const mapDispatchToProps = {
  fetchUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(User);
