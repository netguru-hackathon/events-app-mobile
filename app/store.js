import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';

import axiosClient from './api/client';
import Reducers from './reducers/index';

const store = applyMiddleware(
  thunk,
  axiosMiddleware(axiosClient, { returnRejectedPromiseOnError: true }),
)(createStore)(Reducers);

export default store;
