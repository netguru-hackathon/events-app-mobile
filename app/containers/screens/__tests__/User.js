import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import User from '../User';
import store from '../../../store';

test('renders correctly', () => {
  const tree = renderer.create(
    <User
      store={store}
      isStarted={'true'}
      isFetching={'true'}
      event={{
        avatarUrl: 'avatarUrl',
        email: 'email',
        name: 'name',
      }}
      navigation={{
        navigate: () => {},
        state: {
          params: {
            item: {
              id: 1,
              name: 'name',
            },
          },
        },
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
