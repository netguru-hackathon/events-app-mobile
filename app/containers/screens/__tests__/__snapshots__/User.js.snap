// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "height": 80,
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
</View>
`;
